lib/kbuildbarn/protoparse_test.go

```diff
@@ -0,0 +1,60 @@
+package kbuildbarn_test
+
+import (
+	"github.com/enfabrica/enkit/lib/kbuildbarn"
+	bespb "github.com/enfabrica/enkit/third_party/bazel/buildeventstream"
+	"github.com/stretchr/testify/assert"
+	"testing"
+)
+
+func TestEmpty(t *testing.T) {
+	result := kbuildbarn.GenerateLinksForNamedSetOfFiles(&bespb.NamedSetOfFiles{}, "", "", "")
+	assert.Nil(t, result)
+}
+
+func TestSingleContain(t *testing.T) {
+	simple := &bespb.NamedSetOfFiles{Files: []*bespb.File{
+		{
+			Name: "simple.txt", Digest: "digest", Length: 614,
+		},
+	}}
+	result := kbuildbarn.GenerateLinksForNamedSetOfFiles(simple, "/enfabrica/mymount", "myInvocation", "localCluster")
+	assert.Equal(t, "/enfabrica/mymount/cas/localCluster/blobs/file/digest", result.Src)
+	assert.Equal(t, "/enfabrica/mymount/scratch/myInvocation/simple.txt", result.Dest)
+}
+
+func TestParseMany(t *testing.T) {
+	many := &bespb.NamedSetOfFiles{Files: []*bespb.File{
+		{
+			Name: "simple.txt", Digest: "digest0", Length: 614,
+		},
+		{
+			Name: "hello/simple.txt", Digest: "digest1", Length: 43,
+		},
+		{
+			Name: "one/two/foo.bar", Digest: "digest2", Length: 888,
+		},
+		{
+			Name: "tarball.tar", Digest: "digest3", Length: 777,
+		},
+	}}
+	baseName := "/foo/bar"
+	clusterName := "duster"
+	invocationPrefix := "invocation"
+	expected := map[string]string{
+		"/foo/bar/scratch/invocation/simple.txt":       "/foo/bar/cas/duster/blobs/file/digest0",
```

Comment:
Any advantage to generating relative symlinks?

Reply:
Not really, we have 0 clarity as to where the commands to create symlinks gets run etc. absolute paths are much safer, especially for cleanup.